hazelcast/src/main/java/com/hazelcast/query/impl/predicates/PartitionPredicateImpl.java

```diff
@@ -35,11 +41,13 @@
  * @param <V> type of the entry value
  */
 @BinaryInterface
-public class PartitionPredicateImpl<K, V> implements PartitionPredicate<K, V>, IdentifiedDataSerializable {
+public class PartitionPredicateImpl<K, V> implements PartitionPredicate<K, V>, IdentifiedDataSerializable, Versioned {
 
     private static final long serialVersionUID = 1L;
+    private static final Random RANDOM = new Random();
 
     private Object partitionKey;
```

Comment:
can we delete `partitionKey` and rely on `partitionKeys` in all relevant places?

Reply:
Yes that sounds good to me. I wasn't sure if we wanted a fast path or not.